src/main/java/uk/gov/hmcts/reform/sscs/service/exceptions/FeignClientErrorDecoder.java

```diff
@@ -66,28 +68,37 @@ private HmcFailureMessage extractFailMsg(String methodKey, Response response) {
                 log.error("Error details: {}", new String(originalRequest.body(), StandardCharsets.UTF_8));
             }
             if (payload != null) {
-                failMsg = HmcFailureMessage.builder()
-                    .requestType(httpMethod.toString())
-                    .caseID(Long.valueOf(payload.getCaseDetails().getCaseId()))
-                    .timeStamp(LocalDateTime.now())
-                    .errorCode(String.valueOf(response.status()))
-                    .errorMessage(response.reason())
-                    .build();
+                failMsg = buildFailureMessage(httpMethod.toString(), Long.valueOf(payload.getCaseDetails().getCaseId()),
+                    LocalDateTime.now(), String.valueOf(response.status()), getOriginalErrorMessage(response));
             }
         } else {
             Long caseId = Long.parseLong(response.request().requestTemplate().queries().get("id").iterator().next());
```

Comment:
Would be better to to put this into a method of it's own like getQueryId(response), then no need to set it to a variable too